### PR TITLE
[online-cmaq] Adding changes to use GFS driven biogenics

### PR DESCRIPTION
**Pull request by bbakernoaa**
### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [ ] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [x] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DOCUMENTATION:

This creates the ability to drive the MEGAN biogenics if GFS data is available.  Note that if GFSv16 isn't available it will default back to MERRA2 driven. 

## ISSUE: 

realated to https://github.com/ufs-community/ufs-srweather-app/issues/518
related to https://github.com/NOAA-EMC/AQM/issues/12

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

## CONTRIBUTORS (optional): 

@zmoon @bbakernoaa

### CODE REVIEW

File: ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2 [hunk 1]
```diff
@@ -82,5 +82,6 @@ cpl_aqm_parm:
   NEXUS_INPUT_DIR: /lfs/h2/emc/lam/noscrub/RRFS_CMAQ/nexus_emissions
   NEXUS_FIX_DIR: /lfs/h2/emc/lam/noscrub/RRFS_CMAQ/nexus/fix
   NEXUS_GRID_FN: grid_spec_793.nc
+  RUN_TASK_NEXUS_GFS_SFC: true
```
Comment (chan-hoo) [inline ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2]: @bbakernoaa , this is set on Line 55 above, please remove this line

File: ush/sample_config/ARL_run/config.yaml_community_aqmna13km_1day_hera [hunk 2]
```diff
@@ -94,3 +94,4 @@ cpl_aqm_parm:
   NUM_SPLIT_NEXUS: 1
   RESTART_WORKFLOW: true
   RESTART_CYCLE_DIR: /scratch2/NCEPDEV/stmp3/Jianping.Huang/NCO_OUTPUT/stmp/tmpnwprd/aqm_test_a/2020091412
+  RUN_TASK_NEXUS_GFS_SFC: true
```
Comment (chan-hoo) [inline ush/sample_config/ARL_run/config.yaml_community_aqmna13km_1day_hera]: Same as above. Please remove this line.

File: ush/sample_config/ARL_run/config.yaml_community_conus13km_1day_hera [hunk 3]
```diff
@@ -83,3 +83,4 @@ cpl_aqm_parm:
   NUM_SPLIT_NEXUS: 1
   COLDSTART: false
   WARMSTART_CYCLE_DIR: /scratch2/NCEPDEV/stmp3/Jianping.Huang/expt_dirs/uwm_aqm_cs_canopy_a/2019073112
+  RUN_TASK_NEXUS_GFS_SFC: true
```
Comment (chan-hoo) [inline ush/sample_config/ARL_run/config.yaml_community_conus13km_1day_hera]: Sam as above. Please remove this line.

File: ush/sample_config/EMC_run/config.yaml_nco_aqmna13km_1day_wcoss2 [hunk 4]
```diff
@@ -104,5 +104,6 @@ cpl_aqm_parm:
   NEXUS_GRID_FN: grid_spec_793.nc
   NUM_SPLIT_NEXUS: 3
   AQM_AIRNOW_HIST_DIR: /lfs/h2/emc/physics/noscrub/kai.wang/Bias_correction/aqmv7.0
+  RUN_TASK_NEXUS_GFS_SFC: true
```
Comment (chan-hoo) [inline ush/sample_config/EMC_run/config.yaml_nco_aqmna13km_1day_wcoss2]: Same as above. Please remove this line.

File: ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2 (hunk 1 above)
Comment (bbakernoaa) [reply ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2]: why are we turning this off?

Comment (bbakernoaa) [reply ush/sample_config/ARL_run/config.yaml_cold_aqmna13km_1day_wcoss2]: ok

**Comment by chan-hoo** [comment]
@bbakernoaa, I got the following error with your branch:  `/lfs/h2/emc/lam/save/chan-hoo.jeon/online-cmaq_test/ufs-srweather-app/scripts/exregional_nexus_emission.sh: line 272: ./nexus_gfs_bio.py: Permission denied`. Please change the permission of this file.

**Comment by bbakernoaa** [comment]
please pull the latest develop branch of nexus.  I'll update the hash for the workflow

**Comment by chan-hoo** [comment]
@bbakernoaa, I got another error: ./nexus_gfs_bio.py: line 4: $'\nExtract variables from GFS output and format for HEMCO\n(i.e., as if MERRA-2)\n': command not found
./nexus_gfs_bio.py: line 5: from: command not found
./nexus_gfs_bio.py: line 41: M2_DS_ATTRS: command not found
./nexus_gfs_bio.py: line 49: Format:: command not found
./nexus_gfs_bio.py: line 50: SpatialCoverage:: command not found
./nexus_gfs_bio.py: line 51: Conventions:: command not found
./nexus_gfs_bio.py: line 60: Delta_Lon:: command not found
./nexus_gfs_bio.py: line 61: Delta_Lat:: command not found
./nexus_gfs_bio.py: line 62: syntax error near unexpected token `}'
./nexus_gfs_bio.py: line 62: `}'

**Comment by bbakernoaa** [comment]
forgot to add the shebang to the script.  Please update to the hash [69a7a50](https://github.com/noaa-oar-arl/NEXUS/commit/69a7a508a8d52afab5c05a6f121d7f9517668e6a).  You shouldn't need to recompile

File: externals/Externals_AQM.cfg [hunk 5]
```diff
@@ -12,7 +12,7 @@ protocol = git
 repo_url = https://github.com/noaa-oar-arl/NEXUS
 # Specify either a branch name or a hash but not both.
 #branch = develop
-hash = 5c1fa37
+hash = 2314ffe
```
Comment (chan-hoo) [inline externals/Externals_AQM.cfg]: @bbakernoaa , can you update this hash?

**Comment by chan-hoo** [comment]
It looks working well now. approving

**Review by chan-hoo — APPROVED** [review]

**Pull request merged**